docs-src/tcp.md

```diff
@@ -0,0 +1,23 @@
+Service APIs is designed to work with multiple protocols (unlike the HTTP only
+Ingress/v1 resource). [TCPRoute](spec/#networking.x-k8s.io/v1alpha1.TCPRoute) is
+one such route which allows for managing TCP traffic.
+
+In this example, we have one Gateway resource and two TCPRoute resources that
+distribute the traffic with the following rules:
+
+- All TCP streams on port 8080 of the Gateway are forwarded to port 6000 of
+  `my-foo-service` Kubernetes Service.
+- All TCP streams on port 8090 of the Gateway are forwarded to port 6000 of
+  `my-bar-service` Kubernetes Service.
+
+Please note the following:
+
+- The `protocol` of listeners on the Gateway is `TCP`.
+- Each listener selects exactly one TCPRoute. This is important since routing
+  decision is performed based on destination port only. If more metadata is used
+  for routing decisions, then one may associate multiple TCPRoutes to a single
+  Gateway listener.
```

Comment:
I think it would be helpful to describe briefly what this may look like. Maybe something like:

```
Implementations may support custom resources for more advanced matching logic. This could be used to associate multiple TCPRoutes with a single Gateway listener.
```

Reply:
Worded this a little differently. PTAL.